test/integration/mxGraph.model.style.api.test.ts

```diff
@@ -625,3 +625,272 @@ describe('mxGraph model - update style', () => {
+        // not under test
+        parentId: 'lane_03',
+        verticalAlign: 'bottom',
+      });
+      expect('sequenceFlow_lane_1_elt_1').toBeSequenceFlow({
+        // not under test
+        parentId: 'lane_01',
+        verticalAlign: 'bottom',
+      });
+    });
+
+    it('Reset the style twice', () => {
+      const strokeColor = 'DarkBlue';
+      bpmnVisualization.bpmnElementsRegistry.updateStyle(['sequenceFlow_lane_1_elt_1'], { stroke: { color: strokeColor } });
+      bpmnVisualization.bpmnElementsRegistry.updateStyle(['sequenceFlow_lane_1_elt_1'], {});
+
+      expect('sequenceFlow_lane_1_elt_1').toBeSequenceFlow({
+        stroke: { color: strokeColor },
+        // not under test
+        parentId: 'lane_01',
+        verticalAlign: 'bottom',
+      });
+    });
+  });
+
+  describe('Both Edges and Shapes', () => {
+    beforeEach(() => {
+      bpmnVisualization.load(readFileSync('../fixtures/bpmn/registry/1-pool-3-lanes-message-start-end-intermediate-events.bpmn'));
+    });
+
+    it('Non existing elements', () => {
+      // Check that there is no error
+      bpmnVisualization.bpmnElementsRegistry.resetStyle(['i_do_not_exist_1', 'i_do_not_exist_2']);
+    });
+
+    it('Several elements', () => {
+      const elementIds = ['startEvent_lane_1', 'sequenceFlow_lane_1_elt_1'];
+
+      // Apply custom style
+      const customStyle = {
+        stroke: { color: 'orange' },
+      };
+      bpmnVisualization.bpmnElementsRegistry.updateStyle(elementIds, customStyle);
+
+      // Reset style
+      bpmnVisualization.bpmnElementsRegistry.resetStyle(elementIds);
+
+      // Check that the style has been reset to default values for each element
+      expect('startEvent_lane_1').toBeStartEvent({
+        // not under test
+        eventDefinitionKind: ShapeBpmnEventDefinitionKind.MESSAGE,
+        parentId: 'lane_01',
+        label: 'message start 1',
+      });
+      expect('sequenceFlow_lane_1_elt_1').toBeSequenceFlow({
+        // not under test
+        parentId: 'lane_01',
+        verticalAlign: 'bottom',
+      });
+    });
+  });
+
+  // Check that there is no bad interactions between the two features
+  describe('Style API Reset and CSS class update', () => {
+    it.each([true, false])('Reset style first %s', (isStyleResetFirst: boolean) => {
+      bv.load(readFileSync('../fixtures/bpmn/registry/1-pool-3-lanes-message-start-end-intermediate-events.bpmn'));
+
+      const bpmnElementId = 'endEvent_message_1';
+      const cssClassName = ['class-1', 'class-2'];
+
+      bv.bpmnElementsRegistry.updateStyle(bpmnElementId, { stroke: { color: 'pink' } });
+
+      if (isStyleResetFirst) {
+        bv.bpmnElementsRegistry.resetStyle(bpmnElementId);
+        bv.bpmnElementsRegistry.addCssClasses(bpmnElementId, cssClassName);
+      } else {
+        bv.bpmnElementsRegistry.addCssClasses(bpmnElementId, cssClassName);
+        bv.bpmnElementsRegistry.resetStyle(bpmnElementId);
```

Comment:
**question:**  @tbouffard 
If the reset occurs after adding CSS classes in the current implementation, it fails because adding CSS classes triggers the same event as updateStyle: MxStyleChange. This event enables the recording of the previous style state for later resetting.

Should we retain this implementation as is, or do we need to distinguish between the events?

Reply:
In order to proceed, I need the PR typed-mxgraph/typed-mxgraph#53 to be merged, and the repository [https://github.com/typed-mxgraph/typed-mxgraph](https://github.com/typed-mxgraph/typed-mxgraph) to be released.   
I tried to differentiate the two events, but these actions are necessary for me to continue. 😊

Reply:
Despite using a custom event, I still encounter the identical problem. Upon reflection, the problem stems from the fact that the `updateStyle` function is called before the `addCssClasses` function. Consequently, when we reset, we revert to the style state before the CSS classes were added.

To address this, we need to change the way we store the style. In addition, we need to store the CSS classes separately in order to reconstruct the style during the `reset` process.

Reply:
As discuss together, we won't merge this PR as it is now.
Feel free to create a new PR or update this one with the implementation that lets handle the CSS classes